crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -222,17 +229,21 @@ module sui::staking_pool {
     /// Called at epoch boundaries to process pending delegation withdraws requested during the epoch.
     /// For each pending withdraw entry, we withdraw the rewards from the pool at the new exchange rate and burn the pool
     /// tokens.
-    public(friend) fun process_pending_delegation_withdraws(pool: &mut StakingPool, new_epoch: u64, ctx: &mut TxContext) : u64 {
+    public(friend) fun process_pending_delegation_withdraws(
+        pool: &mut StakingPool, processing_limit: u64, new_epoch: u64, ctx: &mut TxContext
+    ) : u64 {
         let total_reward_withdraw = 0;
-
-        while (!table_vec::is_empty(&pool.pending_withdraws)) {
-            let PendingWithdrawEntry {
+        let num_withdraws_processed = 0;
+        // Process pending withdraws up to the provided limit.
+        while (num_withdraws_processed < processing_limit && !linked_table::is_empty(&pool.pending_withdraws)) {
```

Comment:
Do we want some sort of externally visible indicator that we stopped early because we hit the processing limit (e.g., an event that indicates the # remaining)?